contracts/drafts/dao/DAO.sol

```diff
@@ -0,0 +1,157 @@
+pragma solidity ^0.5.10;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20Mintable.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "../../issuance/IssuanceEth.sol";
+import "../token/ERC20MultiDividendable.sol";
+
+
+/**
+ * @title DAO
+ * @notice This is an exeprimental DAO (Decentralised Autonomous Organization) implementation. Use with caution.
+ * @dev The contract inhertis from ERC20Mintable and IssuanceEth, so it's an issuance that has its own address as issuance token. It also inherits from ERC20MultiDividendable, so it can invest in other IssuanceEth's issuance tokens and distribute them to the investors.
+ */
+contract DAO is ERC20Mintable, ERC20MultiDividendable, IssuanceEth {
+
+    using SafeMath for uint256;
+
+    mapping(address => uint256) public begAmount;
+    uint256 totalAmounts;
+    mapping(address =>
+        mapping(address => uint256)
+    ) public votesForIdeaByHolder;
+    mapping(address => uint256) public totalVotesForIdea;
+    mapping(address => uint256) public totalVotesByHolder;
+    mapping(address => address[]) public backersForIdea;
+    event Here();
+
+    constructor()
+    ERC20Mintable()
+    ERC20MultiDividendable()
+    IssuanceEth(address(this))
+    public
+    {
+        _createState("NEVER");
+        _createTransition("LIVE", "SETUP");
```

Comment:
Are these transitions to "SETUP" so that the issuePrice for DAO tokens can be redefined?

Reply:
These transitions are so that at any time, new investors can put up ether into the DAO and receive back DAO tokens. This is done through the restartFundingRound function.

Reply:
Yes, you are right, I will change this.

Reply:
Actually, now I think it's to better with `"SETUP"`. The owner of the `DAO` might want to increase/decrease `issuePrice`, depending on the success of the `DAO` investments.

Reply:
Mmm, the owner of the DAO are the DAO stakeholders... We are going down some rabbit hole here.

Reply:
Yes, they are, but by the `owner` I meant here the literal `owner`, i.e. the address of the wallet that deployed the contract. So, yes, you are right, and yes, I am also right.

Reply:
I've just seen how you dealt with the dividends collected from funded ventures, and now everything makes a lot of sense to me. Let's discuss this carefully.

I think that the DAO should vote not only on funding proposals, but also on corporate actions. I don't think that there should be a DAO owner with any power or responsibility.

Even in this simple DAO there are two corporate actions I can see:
1. Additional funding rounds: It should be voted when to have them and with which parameters (issue price, start date, end date, etc).
2. Disbursement of DAO dividends. Here you solve the issue by distributing all dividends from funded ventures as DAO dividends. I think that venture dividends should join the DAO pool. When and how much of the pool to disbursed to DAO stakeholders should be a corporate action and should be voted.

I would suggest we:
 - Remove the possiblity of additional funding rounds from this DAO.
 - Leave the dividend disbursement mechanism as you coded it.
 - Make `withdraw` internal.

That would make the DAO truly decentralized and simple enough to understand. We could merge it straight into `examples` as a DAO VC Fund.

Then in further contracts we should see how to generalize the DAO one step so that there are corporate actions that can get voted, with the corporate actions being things like funding ventures, disbursing dividends, or issuing new tokens.

What are your thoughts?